PWGHF/Tasks/taskBPlus.cxx

```diff
@@ -114,66 +144,14 @@ struct HfTaskBplus {
       }
     } // candidate loop
   }   // process
-};    // struct
 
-/// BPlus MC analysis and fill histograms
-struct HfTaskBplusMc {
-  HistogramRegistry registry{
-    "registry",
-    {{"hPtRecSig", "B+ candidates (matched);candidate #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{300, 0., 30.}}}},
-     {"hPtRecBg", "B+ candidates (unmatched);candidate #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{300, 0., 30.}}}},
-     {"hPtGenSig", "B+ candidates (matched);candidate #it{p}_{T}^{gen.} (GeV/#it{c});entries", {HistType::kTH1F, {{300, 0., 10.}}}},
-     {"hPtGen", "MC particles (matched);candidate #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{300, 0., 30.}}}}}};
+  Partition<soa::Join<aod::HfCandBPlus, aod::HFSelBPlusToD0PiCandidate, aod::HfCandBPMCRec>> recoFlagBPlus = aod::hf_selcandidate_bplus::isSelBPlusToD0Pi >= selectionFlagBPlus;
```

Comment:
`recoFlagBPlus` is a weird name for selected candidates. You can use something like `selectedBPlusCandidatesMC`. (I see this appears in other tasks as well.)